packages/omgidl-serialization/src/DeserializationInfoCache.ts

```diff
@@ -209,11 +232,99 @@ export class DeserializationInfoCache {
       name,
       type,
       typeDeserInfo: fieldDeserInfo,
+      isComplex: false,
       isArray,
       arrayLengths,
       definitionId: getDefinitionId(definition),
+      isOptional: isOptional(definition),
     };
   }
+
+  public getFieldDefault(deserInfo: FieldDeserializationInfo): unknown {
```

Comment:
Do we need this function? Can't we pre-compute the default values when creating the deserialization info cache objects? 

E.g. I would expect to be able to get the default value with `deserInfo.defaultValue` instead of calling `getFieldDefault(deserInfo)` at message deserialization time

Reply:
I didn't want to precompute the default values if they weren't needed to prevent unnecessary computation and memory usage, but if that intuition is wrong we can precompute.

Reply:
Yeah I think it would be better to pre-compute default values for faster message deserialization. I don't think that we have to worry about memory usage much, it should just be a few bytes per field (for `0`, `""` or `[]`).

Reply:
My only qualm is that these are being created even if they aren't used. Lazily calculating them only has a one-time cost and then they are saved. I guess this is a bit of premature optimization though

Reply:
Lazy calculation is also fine. I guess my main comment was where to store the default values, I find it more intuitive to store them in the deserialization info objects (as optional field if computed lazily) rather than in `#complexDefaultValueGetters`. This should avoid a map lookup

Reply:
ah yeah that definitely makes more sense. will update then.